src/sandstorm/spk.c++

```diff
@@ -918,12 +1006,66 @@ private:
+          "  #  # https://github.com/sandstorm-io/sandstorm/blob/master/src/sandstorm/package.capnp\n"
+          "  #  viewInfo = (\n"
+          "  #    # For details on the viewInfo field, consult \"ViewInfo\" in\n"
+          "  #    # https://github.com/sandstorm-io/sandstorm/blob/master/src/sandstorm/grain.capnp\n"
+          "  #\n"
+          "  #    permissions = [\n"
+          "  #    # Permissions which a user may or may not possess.  A user's current\n"
+          "  #    # permissions are passed to the app as a comma-separated list of `name`\n"
+          "  #    # fields in the X-Sandstorm-Permissions header with each request.\n"
+          "  #    # See \"PermissionDef\" in \n"
+          "  #    # https://github.com/sandstorm-io/sandstorm/blob/master/src/sandstorm/grain.capnp\n"
+          "  #      (\n"
+          "  #        name = \"editor\",\n"
+          "  #        # Name of the permission, used as an identifier for the permission in cases where string\n"
+          "  #        # names are preferred.  Used in sandstorm-http-bridge's X-Sandstorm-Permissions HTTP header.\n"
+          "  #\n"
+          "  #        title = (defaultText = \"editor\"),\n"
+          "  #        # Display name of the permission, e.g. to display in a checklist of permissions\n"
+          "  #        # that may be assigned when sharing.\n"
+          "  #\n"
+          "  #        description = (defaultText = \"grants ability to modify data\"),\n"
+          "  #        # Prose describing what this role means, suitable for a tool tip or similar help text.\n"
+          "  #      ),\n"
+          "  #    ],\n"
+          "  #    roles = [\n"
+          "  #      # Roles are logical collections of permissions.  For instance, your app may have\n"
+          "  #      # a \"viewer\" role and an \"editor\" \n"
+          "  #      (\n"
+          "  #        title = (defaultText = \"admin\"),\n"
+          "  #        # Name of the role.  Shown in the Sandstorm UI to indicate which users have which roles.\n"
+          "  #\n"
+          "  #        permissions  = [true],\n"
+          "  #        # An array indicating which permissions this role carries.\n"
+          "  #        # It should be the same length as the permissions array in\n"
+          "  #        # viewInfo, and the order of the lists must match.\n"
+          "  #\n"
+          "  #        verbPhrase = (defaultText = \"can change site settings\"),\n"
+          "  #        # Brief explanatory text to show in the sharing UI indicating\n"
+          "  #        # what a user assigned this role will be able to do with the grain.\n"
+          "  #\n"
+          "  #        description = (defaultText = \"admins may view all site data and change settings.\"),\n"
+          "  #        # Prose describing what this role means, suitable for a tool tip or similar help text.\n"
+          "  #      ),\n"
+          "  #    ],\n"
+          "  #  ),\n"
+          "  #  #apiPath = \"/api\",\n"
```

Comment:
This is a pretty undocumented thing! I mean within this file. Add a comment maybe saying

```
# Apps can export an API to the world. The API is to be used by Javascript code and
# native apps, so it can't serve out regular HTML to browsers. If a request comes in to your app's API,
# sandstorm-http-bridge will prefix the request's path with this string.
```

something like that to indicate (a) what we mean by API ("oh this is for cap'nproto API exporting?") and (b) that the API not only shouldn't be used to expose your whole app, it _can't_ be effectively used that way ("oh I'll expose the whole app as an API so that I can bypass the Sandstorm sharing complexity")